tests/rbac_test.go

```diff
@@ -0,0 +1,228 @@
+			_, err = ktests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Check VMI should create.")
+			args = []string{"start", vmi.Name, "-n", NamespaceTestSystem}
+			_, err = ktests.RunCommand("virtctl", args...)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Check the vmi is running.")
+			args = []string{"get", "vmi", vmi.Name, "--template", "{{.status.phase}}", "-n", NamespaceTestSystem}
+			Eventually(func() string {
+				output, err := ktests.RunCommand("oc", args...)
+				Expect(err).ToNot(HaveOccurred())
+				return output
+			}, time.Minute*10, time.Second*1).Should(Equal("Running"))
+
+			By("Connect via console.")
+			vm, err := virtClient.VirtualMachineInstance(NamespaceTestSystem).Get(vmi.Name, &metav1.GetOptions{})
+			expecter, err := ktests.LoggedInCirrosExpecter(vm)
+			Expect(err).ToNot(HaveOccurred())
+			expecter.Close()
+
+			By("Checking the response from VNC server.")
+			response, err := tests.VNCConnection(NamespaceTestSystem, vmi.Name)
+			Expect(response).To(Equal("RFB 003.008"))
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Check VMI should stop.")
+			args = []string{"stop", vmi.Name, "-n", NamespaceTestSystem}
+			_, err = ktests.RunCommand("virtctl", args...)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Delete a VM with yaml.")
+			args = []string{"delete", "-f", vmi.Manifest, "-n", NamespaceTestSystem}
+			_, err = ktests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Delete a namespace")
+			args = []string{"delete", "project", NamespaceTestSystem}
+			_, err = ktests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	DescribeTable("RBAC Authorization", func(role, namespace string, con bool) {
+		By(fmt.Sprintf("Add %s to user %s", role, tests.UsernameTestUser))
+		args := []string{"adm", "policy", "add-role-to-user", role, tests.UsernameTestUser, "-n", namespace}
+		_, err := ktests.RunCommand("oc", args...)
+		Expect(err).ToNot(HaveOccurred())
+
+		if con == false {
+			createResourcesToTestViewRole()
+		}
+
+		By(fmt.Sprintf("Login with the user %s", tests.UsernameTestUser))
+		args = []string{"login", "-u", tests.UsernameTestUser, "-p", "123456"}
+		_, err = ktests.RunCommand("oc", args...)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Create a VM with yaml.")
+		args = []string{"create", "-f", vmi.Manifest, "-n", namespace}
+		_, err = ktests.RunCommand("oc", args...)
+		if con == true {
+			Expect(err).ToNot(HaveOccurred())
+		} else {
+			Expect(err).To(HaveOccurred())
+		}
+
+		By("Check VMI should create.")
+		args = []string{"start", vmi.Name, "-n", namespace}
+		_, err = ktests.RunCommand("virtctl", args...)
+		if con == true {
+			Expect(err).ToNot(HaveOccurred())
+		} else {
+			Expect(err).To(HaveOccurred())
+		}
+
+		By("Check the vmi is running.")
+		args = []string{"get", "vmi", vmi.Name, "--template", "{{.status.phase}}", "-n", namespace}
+		result := Eventually(func() string {
```

Comment:
Assigning `Eventually` matcher into a variable is generally not a good coding practice, because it decreases code readability. I'd recommend sticking to the more idiomatic implementation:

```go
getVMIStatus := func() string {
	output, err := ktests.RunCommand("oc", "get", "vmi", vmi.Name, "--template", "{{.status.phase}}", "-n", namespace)
	if con == true {
		ExpectWithOffset(1, err).ToNot(HaveOccurred(), "failed to get VMI status via oc command: %s: %v", output, err)
	} else {
		ExpectWithOffset(1, err).To(HaveOccurred(), "still able to get VMI status via oc command: %s", output)
	}
	return output
}

if con == true {
	Eventually(getVMIStatus, 10*time.Minute, 1*time.Second).Should(Equal("Running"), "VMI %q is still not running", vmi.Name)
}
```